brave/brave-core
# Promotion guards

Resolves https://github.com/brave/brave-browser/issues/9662
Resolves https://github.com/brave/brave-browser/issues/9660

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
- enable rewards
- click on claim promotion (do not solve it)
- open db and change promotion status to 1
- solve a captcha
- make sure that you see an error and status stays at 1
- promotion should be cleared as well

Public key flow is covered with browser tests

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
